### DISCUSSION TITLE
Why does `--verifytypes` want an explicit type annotation on some variables?

### DISCUSSION THREAD

**Discussion by Pyprohly** [microsoft/pyright#2272] [Q&A] [answered]
When using `--verifytypes`, Pyright produces a lot of errors in places I wouldn’t expect.

For example, in:

```python
from typing import NamedTuple

class A(NamedTuple):
    a: int

a = A(1)
```

Pyright says `mypkg/a.py:7:1 - error: Type unknown for variable "mypkg.a.a"`.

Why does it want an explicit type annotation for `a` here?

**Comment by erictraut** [comment, marked as answer]
If your library is marked "py.typed", all public symbols (those that represent the public interface contract for your library) should be annotated. If a symbol is not annotated, its type needs to be inferred by a type checker. Type inference rules are not specified in the typing standards, and they vary by type checker. Inference can also be quite slow depending on the complexity of the expression. For these reasons, it's important for all public symbols to have type annotations.

For more details, please refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#type-completeness).
